File: Assets/Bundles/Client/Resources/TS/Controllers/Global/Camera/CameraController.lua
-- Compiled with unity-ts v2.1.0-75
local Reflect = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Reflect
local Controller = require("Shared/rbxts_include/node_modules/@easy-games/flamework-core/out/init").Controller
local CameraSystem = require("Client/TS/Controllers/Global/Camera/CameraSystem").CameraSystem
local CameraController
do
	CameraController = setmetatable({}, {
		__tostring = function()
			return "CameraController"
		end,
	})
	CameraController.__index = CameraController
	function CameraController.new(...)
		local self = setmetatable({}, CameraController)
		return self:constructor(...) or self
	end
	function CameraController:constructor()
		self.mainCamera = GameObject:Find("MainCamera"):GetComponent("Camera")
		self.cameraSystem = CameraSystem.new(self.mainCamera)
	end
	function CameraController:SetMode(mode, transition)
		self.cameraSystem:SetMode(mode, transition)
	end
	function CameraController:ClearMode(transition)
		self.cameraSystem:ClearMode(transition)
	end
	function CameraController:SetFOV(fieldOfView, immediate)
		if immediate == nil then
			immediate = false
		end
		self.cameraSystem:SetFOV(fieldOfView, immediate)
	end
	function CameraController:OnStart()
	end
	CameraController.CameraReferenceKey = "CameraRig"
end
-- (Flamework) CameraController metadata
Reflect.defineMetadata(CameraController, "identifier", "Bundles/Client/Controllers/Global/Camera/CameraController@CameraController")
Reflect.defineMetadata(CameraController, "flamework:implements", { "$:flamework@OnStart" })
Reflect.decorate(CameraController, "$:flamework@Controller", Controller, { {} })
return {
	CameraController = CameraController,
}
-- ----------------------------------
-- ----------------------------------

File: Assets/Typescript~/src/Client/Controllers/Global/Camera/CameraController.ts
import { Controller, OnStart } from "@easy-games/flamework-core";
import { CameraMode, CameraModeTransition } from ".";
import { CameraSystem } from "./CameraSystem";
import { BundleReferenceManager } from "../../../../Shared/Util/BundleReferenceManager";

@Controller({})
export class CameraController implements OnStart {
	public static readonly CameraReferenceKey: string = "CameraRig";

	/** The underlying camera system for the game. */
	public readonly cameraSystem: CameraSystem;
	public readonly mainCamera: Camera;

	constructor() {
		this.mainCamera = GameObject.Find("MainCamera").GetComponent<Camera>();
		this.cameraSystem = new CameraSystem(this.mainCamera);
	}

	/**
	 * Proxy for `cameraSystem.SetMode()`.
	 *
	 * Set the current camera mode. If `transition` is provided, then the new
	 * mode will be interpolated from the old mode based on the configuration
	 * provided within `transition`. Otherwise, the camera will snap immediately
	 * to the new mode.
	 *
	 * @param mode New mode.
	 * @param transition Optional transition configuration.
	 */
	public SetMode(mode: CameraMode, transition?: CameraModeTransition) {
		this.cameraSystem.SetMode(mode, transition);
	}

	/**
	 * Proxy for `cameraSystem.ClearMode()`.
	 *
	 * Sets the camera to a static view.
	 *
	 * @param transition Optional transition configuration.
	 */
	public ClearMode(transition?: CameraModeTransition) {
		this.cameraSystem.ClearMode(transition);
	}

	/**
	 * Proxy for `cameraSystem.SetFOV()`.
	 *
	 * Set the camera's field-of-view.
	 * @param fieldOfView Field of view.
	 * @param immediate If `true`, goes directly to the FOV without springing towards it.
	 */
	public SetFOV(fieldOfView: number, immediate = false) {
		this.cameraSystem.SetFOV(fieldOfView, immediate);
	}

	OnStart() {}
}
